hyper/http11/connection.py

```diff
@@ -117,6 +122,7 @@ def connect(self):
                 port = self.proxy_port
 
             sock = socket.create_connection((host, port), 5)
```

Comment:
We probably want to use the timeout in `create_connection` too: see [the docs](https://docs.python.org/2/library/socket.html#socket.create_connection) for how to do it.